x/contracts/examples/multisig/src/lib.rs

```diff
@@ -0,0 +1,468 @@
+use std::{collections::HashSet, num::NonZeroUsize};
+
+use wasmlanche::{
+    borsh::{BorshDeserialize, BorshSerialize},
+    public, state_schema, Address, Context,
+};
+
+type ProposalId = usize;
+
+// TODO: make a state_keys module so I can reuse `Proposal`.
+state_schema! {
+    ProposalCount => ProposalId,
+    ProposalItem(ProposalId) => Proposal,
+    ProposalVoteInfo(ProposalId) => ProposalMeta,
+    Vote(ProposalId, Address) => bool,
+}
+
+#[public]
+pub fn propose(ctx: &mut Context, proposal: Proposal, meta: ProposalMeta) -> ProposalId {
+    let actor = ctx.actor();
+
+    let id = ctx
+        .get(ProposalCount)
+        .expect("failed to deserialize")
+        .unwrap_or_default();
+
+    ctx.store((
+        (ProposalCount, id + 1),
+        (ProposalItem(id), proposal),
+        (ProposalVoteInfo(id), meta),
+        (Vote(id, actor), true),
+    ))
+    .expect("failed to store proposal");
+
+    id
+}
+
+#[public]
+pub fn vote(ctx: &mut Context, proposal_id: ProposalId, vote: bool) -> Option<Vec<u8>> {
```

Comment:
🤔 good point

Reply:
I'm going to leave the `Any` variant as it makes the logic more interesting. Addresses are cheap to create, but they aren't quite free. So you wouldn't use that for anything where you wanted Sybil-attack protection... but this is just an example. 

There are two checks, but the checks are fundamentally different. If I removed voters from the set, it could drastically change the price between the 1st vote and the last vote (or really the 2nd to last vote). The first check is for the double vote. The second check is to see whether the person is even allowed to vote. 

It's a weak justification I know... but this is just an example after all.

Reply:
I'm cool with that!